Update getting started guide

cllns opened the pull request:
The main change I wanted to make here was that I realized our spec [where we introduce dynamic data was a false positive](https://github.com/hanami/guides/blob/main/content/v1.3/introduction/getting-started.md#displaying-dynamic-data). That's a bad practice for Behavior Driven Development, so I changed the spec to be a single book, and check for the actual content, rather than a CSS selector and count. This is more robust.

I also changed the link from "Clean architecture" to "Hexagonal architecture". The latter has a wikipedia page and it gets the underlying idea across well enough. (Plus we want to distance ourselves from Robert Martin, who drafted the post we are currently linking too.)

In the process, I went through the whole guide and cleaned up many pretty small things.

adam12 commented on content/v1.3/introduction/getting-started.md:
````diff
@@ -1133,9 +1137,22 @@ We can use the `routes` helper method that is available in our views and actions
+...
+```
+
+(You could also extend this to the specs if you'd like,
````
Do you think we need to parenthesis this statement?

cllns replied:
Removed parenthesis, and added _why_ that might be prefererd.

"You could also extend this to the specs if you'd like, but changing the path can affect users so you may want to repeat yourself there (to make a potentially breaking change for users harder to implement)."

I don't _love_ that this is the note we end on before the "Wrapping Up" section, since it's not directly related to Hanami.

adam12 commented on content/v1.3/introduction/getting-started.md:
````diff
@@ -491,12 +493,15 @@ We can verify it all works as expected with a unit test:
 end
 ```
 
+(Generally we recommend against 'testing the framework' like this, but it's useful here to demonstrate how Entities work in Hanami.)
````
I'm not sure I feel this needs to be parenthesized.

cllns replied:
I changed this to a "convention" block, which seems appropriate. It's a (Green) aside now, which feels right. Users can test the framework if they want! I just wouldn't :)

adam12 replied:
> It's a (Green) aside now, which feels right.

I'm not seeing this in the preview. Was this something you saw locally?

cllns replied:
Whoops, I had a typo :) https://github.com/hanami/guides/pull/93/commits/05d5f5c413ef228802eda843d18c8fb2fa61d7cd

cllns commented on content/v1.3/introduction/getting-started.md:
```diff
@@ -549,8 +554,10 @@ RSpec.describe 'List books' do
+We create a single Book record in our test and then assert that the title and author are displayed on the page.
+When we run this test, it will fail.
+This is because we haven't updated our template yet,
+it still has the hard-coded books from earlier.
```
Updated this to "We create a single Book record in our test and then expect that the title and author are displayed on the page. This test fails since we haven't updated our template yet, and it still includes the hard-coded books from earlier."

What do you think of that?

cllns commented:
Thanks for the careful and thoughful review @adam12 :)

I think I addressed the concerns. Let me know about the items I left "open" (instead of resolved). Open to more tweaks too

cllns commented:
@adam12 Can you check this again when you get a chance? I added a couple other small commits

cllns commented:
Found and [fixed](https://github.com/hanami/guides/commit/7fadff3788d10be060564bdeeb846e5cc6d4cc3b) an error post-merge 🙃
